### PR TITLE
SONAR-6723 Use SQ version in cache keys

### PR DESCRIPTION
**Pull request by dbmeneses**

### CODE REVIEW

**Comment by henryju** [comment]
Some ITs are failing

File: sonar-batch/src/main/java/org/sonar/batch/bootstrap/PersistentCacheProvider.java [hunk 1]
```diff
@@ -36,9 +42,22 @@ public PersistentCache provide(UserProperties props) {
...
   }
+
+  private String getVersion() {
+    InputStream is = this.getClass().getClassLoader().getResourceAsStream("sq-version.txt");
+    if (is == null) {
+      return null;
+    }
+    try (BufferedReader br = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))) {
+      return br.readLine();
+    } catch (IOException e) {
```
Comment (SonarTech) [inline sonar-batch/src/main/java/org/sonar/batch/bootstrap/PersistentCacheProvider.java]: CRITICAL Either log or rethrow this exception. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1166)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* CRITICAL 1 critical

Watch the comments in this conversation to review them.

**Comment by henryju** [comment]
LGTM

**Comment by dbmeneses** [comment]
merged

**Pull request closed without merging**